### PR TITLE
#6 Added category domain model

### PR DESCRIPTION
**Pull request by rubvardanyan**
Added category domain model

### CODE REVIEW

File: blog/persistence/src/main/kotlin/com/excref/kotblog/blog/persistence/category/CategoryRepository.kt [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import org.springframework.stereotype.Repository
+
+/**
+ * @author Ruben Vardanyan
+ * @since 06/07/2017 12:33
+ */
+@Repository
+interface CategoryRepository : CrudRepository<Category, Long> {
+    /**
+     * Finds tag by name
```
Comment (Biacode) [inline blog/persistence/src/main/kotlin/com/excref/kotblog/blog/persistence/category/CategoryRepository.kt]: Update documentation from `tag` to `category`

File: blog/persistence/src/main/kotlin/com/excref/kotblog/blog/persistence/category/CategoryRepository.kt [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+     *
+     * @param   name The tag name
+     * @return  null if not found
+     */
+    fun findByName(name: String): Category?
+
+    /**
+     * Finds tag by uuid
```
Comment (Biacode) [inline blog/persistence/src/main/kotlin/com/excref/kotblog/blog/persistence/category/CategoryRepository.kt]: Here as well

File: blog/service/core/src/main/kotlin/com/excref/kotblog/blog/service/category/CategoryService.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+ * @author Ruben Vardanyan
+ * @since 06/07/2017 12:28
+ */
+interface CategoryService {
+    /**
+     * Creates a new category with the given name
+     *
+     * @param   name The category name
+     * @return  Persistent category
+     * @throws  com.excref.kotblog.blog.service.tag.exception.TagAlreadyExistsForNameException If the category already exists for the given name
```
Comment (Biacode) [inline blog/service/core/src/main/kotlin/com/excref/kotblog/blog/service/category/CategoryService.kt]: change from `TagAlreadyExistsForNameException` to `CategoryAlreadyExistsForNameException`

File: blog/service/core/src/main/kotlin/com/excref/kotblog/blog/service/category/CategoryService.kt [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+     */
+    fun create(name: String): Category
+
+    /**
+     * Gets category by given uuid
+     *
+     * @param   uuid The category uuid
+     * @return  The category
+     * @throws  com.excref.kotblog.blog.service.tag.exception.TagNotExistsForUuidException If category not exists
```
Comment (Biacode) [inline blog/service/core/src/main/kotlin/com/excref/kotblog/blog/service/category/CategoryService.kt]: Here as well

File: blog/service/impl/src/main/kotlin/com/excref/kotblog/blog/service/category/impl/CategoryServiceImpl.kt [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+import org.springframework.transaction.annotation.Transactional
+
+/**
+ * @author Ruben Vardanyan
+ * @since 06/07/2017 12:31
+ */
+@Service
+class CategoryServiceImpl : CategoryService {
+
+    init {
```
Comment (Biacode) [inline blog/service/impl/src/main/kotlin/com/excref/kotblog/blog/service/category/impl/CategoryServiceImpl.kt]: Let's discuss do we really need this garbage code? 😃  I think we can remove it from the project?

File: blog/service/impl/src/test/kotlin/com/excref/kotblog/blog/service/category/CategoryServiceIntegrationTest.kt [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories
+import org.springframework.stereotype.Component
+
+/**
+ * @author Ruben Vardanyan
+ * @since 06/07/2017 12:42
+ */
+@EnableAutoConfiguration
```
Comment (Biacode) [inline blog/service/impl/src/test/kotlin/com/excref/kotblog/blog/service/category/CategoryServiceIntegrationTest.kt:15]: Any issue with just extending `AbstractServiceIntegrationTest` ? why we need those annotations?

File: blog/test-toolkit/src/main/kotlin/com/excref/kotblog/blog/service/test/helper/ServiceImplTestHelper.kt [hunk 7]
```diff
@@ -12,5 +13,7 @@ class ServiceImplTestHelper {
     fun buildTag(): Tag = buildTag(UUID.randomUUID().toString())
 
     fun buildTag(name: String): Tag = Tag(name)
+    fun  buildCategory(): Category = buildCategory(UUID.randomUUID().toString())
```
Comment (Biacode) [inline blog/test-toolkit/src/main/kotlin/com/excref/kotblog/blog/service/test/helper/ServiceImplTestHelper.kt]: I jus found better solution, you can use `default` values for the arguments and avoid second method creation :)
Please refactor tag as well if you can find time for it

File: blog/test-toolkit/src/main/kotlin/com/excref/kotblog/blog/service/test/helper/ServiceIntegrationTestHelper.kt [hunk 8]
```diff
@@ -26,5 +31,13 @@ class ServiceIntegrationTestHelper {
     fun persistTag(name: String): Tag {
         return tagService.create(name)
     }
+
+    fun persistCategory() : Category {
```
Comment (Biacode) [inline blog/test-toolkit/src/main/kotlin/com/excref/kotblog/blog/service/test/helper/ServiceIntegrationTestHelper.kt]: We can apply default values for the arguments here as well

**Review by Biacode — COMMENTED** [review]
Nice work 👍 
Please fix issues below when you have time.

File: blog/service/impl/src/main/kotlin/com/excref/kotblog/blog/service/category/impl/CategoryServiceImpl.kt (hunk 5 above)
Comment (rubvardanyan) [reply blog/service/impl/src/main/kotlin/com/excref/kotblog/blog/service/category/impl/CategoryServiceImpl.kt]: You mean logging during initialization? Not really, it is just a tradition from last java projects 😃

Comment (Biacode) [reply blog/service/impl/src/main/kotlin/com/excref/kotblog/blog/service/category/impl/CategoryServiceImpl.kt]: exactly

File: blog/service/impl/src/test/kotlin/com/excref/kotblog/blog/service/category/CategoryServiceIntegrationTest.kt (hunk 6 above)
Comment (Biacode) [reply blog/service/impl/src/test/kotlin/com/excref/kotblog/blog/service/category/CategoryServiceIntegrationTest.kt:15]: Btw, @GurgenArustamyan are looking at this issue

**Pull request merged**